### ISSUE TITLE
CA2009 should not be raised when changing the generic type

### ISSUE THREAD

**Issue by andrewimcclement** [dotnet/roslyn-analyzers#7374]
### Analyzer

**Diagnostic ID**: [CA2009](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2009): `Do not call ToImmutableCollection on an ImmutableCollection value`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0

### Describe the bug

CA2009 should not be raised when changing the generic type, i.e. `ImmutableList<T1>.ToImmutableList<T2>()` should be valid.

### Steps To Reproduce

```csharp
public interface IFoo {}
public class Foo : IFoo {}

public static class Program
{
    public static void Main()
    {
        var foo = new Foo();
        var list1 = ImmutableList.Create(foo);
        var list2 = list1.ToImmutableList<IFoo>(); // Generates CA2009
    }
}
```

### Expected behavior
CA2009 should only trigger if the generic types are identical.

### Actual behavior
CA2009 triggers even if the generic type is different.

**Comment by andrewimcclement** [comment]
How can I progress this issue? There is a PR with a fix - if there are further changes required to the PR, I am happy to make them.

**Comment by andrewimcclement** [comment]
See https://github.com/dotnet/sdk/pull/50392#issuecomment-3288873140 for proposed resolution.
